### PR TITLE
Check that Web Bluetooth is available

### PR DESCRIPTION
**Pull request by microbit-robert**
Bluetooth can be blocked by system policies and this additional check determines if this is the case.

### CODE REVIEW

**Review by microbit-grace — APPROVED** [review]
Didn't manage to test in iOS, but code-wise it looks good to me.

File: src/script/stores/uiStore.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ import { DeviceRequestStates } from '../microbit-interfacing/MicrobitConnection'
 let text: (key: string, vars?: object) => string;
 t.subscribe(t => (text = t));
 
-export const compatibility = writable<CompatibilityStatus>(checkCompatibility());
+export const compatibility = writable<CompatibilityStatus>(await checkCompatibility());
```
Comment (microbit-matt-hillsdon) [inline src/script/stores/uiStore.ts:23]: How worried should I be that this has regressed browser compatibility by using top-level await?

Comment (microbit-robert) [reply src/script/stores/uiStore.ts:23]: Very broken on Chrome 88: 
`Uncaught SyntaxError: Unexpected reserved word`

Comment (microbit-robert) [reply src/script/stores/uiStore.ts:23]: Making this a draft until resolved.

Comment (microbit-robert) [reply src/script/stores/uiStore.ts:23]: We have another one in i18n.ts:

```
await init({
  fallbackLocale: 'en',
  initialLocale: get(persistantLocale),
  // Needed to format <link> style tags.
  ignoreTag: false,
});
```

Comment (microbit-robert) [reply src/script/stores/uiStore.ts:23]: This is already broken for Chrome 88 due to the await in i18n.ts. We'll come back to this later to attempt to target es2017 and lose both top-level awaits.

**Pull request merged**